quickstart/java/redistest/src/main/java/example/demo/App.java

```diff
@@ -1,25 +1,78 @@
 package example.demo;
 
 import redis.clients.jedis.Jedis;
-import redis.clients.jedis.JedisShardInfo;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
 
 /**
  * Redis test
  *
  */
-public class App 
-{
-    public static void main( String[] args )
-    {
+public class App {
+    private static Object staticLock = new Object();
+    private static JedisPool pool;
+    private static String host;
+    private static int port;
+    private static String password;
+    private static JedisPoolConfig config;
 
-        boolean useSsl = true;
+    // Should be called exactly once during app startup logic.
+    public static void initializeSettings(String host, int port, String password) {
+        App.host = host;
+        App.port = port;
+        App.password = password;
+    }
+
+    public static JedisPool getPoolInstance() {
+        if (pool == null) { // avoid synchronization lock if initialization has already happened
+            synchronized(staticLock) {
+                if (pool == null) { // don't re-initialize if another thread beat us to it.
+                    JedisPoolConfig poolConfig = getPoolConfig();
+                    boolean useSsl = port == 6380 ? true : false;
+                    int db = 0;
+                    pool = new JedisPool(poolConfig, host, port, Protocol.DEFAULT_TIMEOUT, password, db, useSsl);
+                }
+            }
+        }
+        return pool;
+    }
+
+    public static JedisPoolConfig getPoolConfig() {
```

Comment:
This method is similar to the [_RedisClientConfig.java_](https://github.com/Azure-Samples/azure-cache-redis-samples/blob/92e349765f6e88ca53bb338a2bea14cebe3e568c/samples/java/ClientSamples/src/main/java/common/RedisClientConfig.java) file that's part of one of the non-quickstart samples, but this `getPoolConfig()` method that I [borrowed from the gist](https://gist.github.com/JonCole/925630df72be1351b21440625ff2671f#file-redis-samplecode-java-jedispool-java-L46-L68) doesn't list quite as many options.

I was debating on adding all of the options from `RedisClientConfig` to this quickstart and removing the corresponding non-quickstart [_PoolUsage.java_](https://github.com/Azure-Samples/azure-cache-redis-samples/blob/92e349765f6e88ca53bb338a2bea14cebe3e568c/samples/java/ClientSamples/src/main/java/jedis/PoolUsage.java) sample, but there are some pros and cons since adding all the options would make the quickstart a bit more complicated. It's not too important to decide right now. We can always make the adjustment later too.